agent/agent-tooling/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/localstorage/LocalFileCache.java

```diff
@@ -63,9 +64,32 @@ File poll() {
     return persistedFilesCache;
   }
 
-  private static List<File> sortPersistedFiles(File folder) {
+  // load existing files that are not older than 48 hours
+  // this will avoid data loss in the case of app crashes and restarts.
+  private static List<File> loadPersistedFiles(File folder) {
     return FileUtil.listTrnFiles(folder).stream()
         .sorted(Comparator.comparing(File::lastModified))
+        .filter(file -> !isExpired(file, TimeUnit.DAYS.toSeconds(2)))
         .collect(Collectors.toList());
   }
+
+  // files that are older than expiredIntervalSeconds (default 48 hours) are expired
+  static boolean isExpired(File file, long expiredIntervalSeconds) {
+    String name = file.getName();
+    int index = name.indexOf('-');
+    if (index == -1) {
+      logger.debug("unexpected .trn file name: {}", name);
+      return false;
```

Comment:
return true.. treat bad name file as expired and ignore it?

Reply:
ya, that's what I wrote first, but was concerned about old filename format, but I guess we can't load those anyways b/c we changed disk format (adding ikey)?

Reply:
Ya, old format won't work without ikey. i think we can ignore it. otherwise, infinite retry.